auth.js

```diff
@@ -1,43 +1,46 @@
 /*****
-This was originally inspired by examples I found googling for (roughly) "NodeJS code user password hash"
-
-Initial Ver:        https://github.com/justsml/escape-from-callback-mountain/commit/f828e863ee04df5b3f7f4f4307681bb3fde5285a?diff=unified
-All commits:        https://github.com/justsml/escape-from-callback-mountain/commits/master
-PRs Show Refactor:  https://github.com/justsml/escape-from-callback-mountain/pulls?utf8=%E2%9C%93&q=is%3Apr
-
+See Credits & README: https://github.com/justsml/escape-from-callback-mountain
 ******/
-
 const Promise           = require('bluebird')
 const {hashStringAsync} = Promise.promisifyAll(require('./lib/crypto'))
 const {logEventAsync}   = Promise.promisifyAll(require('./lib/log'))
 const {openAsync}       = Promise.promisifyAll(require('./lib/db'))
+const {TimeoutError,
+  ValidationError,
+  NotFoundError }       = require('./errors')
 
-var _openHandle = openAsync(); // FYI: Promises include memoization (caching) built into same API
+const _openHandle = openAsync(); // FYI: Promises include memoization (caching) built into same API
 
 module.exports = {auth};
 
 /* auth is our main function */
-function auth({username, password}) {
+function auth({username, password,
+    _onTimeoutError = errorHandler,
+    _onNotFoundError = errorHandler,
+    _onValidationError = errorHandler}) {
```

Comment:
^^^ New pattern I've been using to make testing/writing controllers/integration easier.